src/libsyntax/feature_gate.rs

```diff
@@ -2578,6 +2582,17 @@ pub fn check_crate(krate: &ast::Crate,
         "async closures are unstable"
     ));
 
+    sess
+        .or_pattern_spans
+        .borrow()
+        .iter()
+        .for_each(|span| gate_feature!(
+            &ctx,
+            or_patterns,
+            *span,
+            "or_patterns syntax is experimental"
```

Comment:
```suggestion
            "or-patterns are experimental"
```
I think we should hyphenate here to make it clearer (and not use the feature gate name for the feature name).

Reply:
Agreed. The rule seems to be to hyphenate compound terms when the first word isn't an adjective or noun (e.g., something like "or" or "where", which can really throw off parsing!).